src/cloud-api-adaptor/pkg/securecomms/ppssh/ppssh.go

```diff
@@ -0,0 +1,304 @@
+	log.Printf("Kubernetes Phase: waiting for client to connect\n")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var peer *sshproxy.SshPeer
+	for peer == nil {
+		nConn, err := listener.Accept()
+		if err != nil {
+			log.Fatal("failed to accept incoming connection (Kubernetes Phase): ", err)
+		}
+		//ctx, cancel := context.WithCancel(context.Background())
+		//defer cancel()
+
+		log.Printf("Kubernetes client connected\n")
+		peer, err = KubernetesSShService(ctx, nConn, kubernetesPhaseConfig)
+		if err != nil {
+			log.Printf("Failed k8sPhase: %s", err)
+			peer = nil
+			continue
+		}
+		peer.AddOutbounds(outbounds)
+		err = peer.AddInbounds(inbounds)
+		if err != nil {
+			log.Printf("Failed addInbounds during k8sPhase: %s", err)
+			return
+		}
+		peer.Wait()
+		log.Printf("KubernetesSShService exiting")
+	}
+}
+
+func attestationPhase(listener net.Listener, inbounds sshproxy.Inbounds, outbounds sshproxy.Outbounds, ppSecrets *PpSecrets) *ssh.ServerConfig {
+	// Singleton - accept an unproven connection for attestation
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var peer *sshproxy.SshPeer
+	for peer == nil {
+		log.Printf("Attastation Phase: waiting for client to connect\n")
+		nConn, err := listener.Accept()
+		if err != nil {
+			log.Fatal("Attastation Phase: failed to accept incoming connection: ", err)
+		}
+
+		log.Printf("Attastation Phase: client connected\n")
+
+		peer, err = AttestationSShService(ctx, nConn)
+		if err != nil {
+			log.Print(err.Error())
+			peer = nil
+		}
+	}
+
+	peer.AddOutbounds(outbounds)
+	if err := peer.AddInbounds(inbounds); err != nil {
+		log.Fatal("Attastation Phase: failed to add Inbounds: ", err)
+	}
+
+	for {
+		log.Printf("Attastation Phase: getting keys from KBS\n")
+		ppSecrets.AddKey(WN_PUBLIC_KEY)
+		ppSecrets.AddKey(PP_PRIVATE_KEY)
+		ppSecrets.Go() // wait for the keys
+		config, err := InitKubernetesPhaseSshConfig(ppSecrets)
+		if err == nil {
+			log.Printf("Attastation Phase: InitKubernetesPhaseSshConfig is ready\n")
+			peer.Upgrade()
+			return config
+		}
+		log.Printf("Attastation Phase: failed getting keys from KBS: %v\n", err)
+	}
+	// wait for a provenPpPrivateKey
+	//WaitForProvenKeys(ctx, peer)
+	//peer.Wait()
+}
+
+func InitSshServer(inbound_strings, outbounds_strings []string, getSecret GetSecret) {
+	log.Printf("Using PP Secure Comms: InitSshServer version %s", sshutil.PpSecureCommsVersion)
+
+	Singleton()
+	var inbounds sshproxy.Inbounds
+	var outbounds sshproxy.Outbounds
+	var wg sync.WaitGroup
+	for _, tag := range inbound_strings {
+		_, _, err := inbounds.Add(tag, &wg)
+		if err != nil {
+			log.Fatalf("failed to add inbound %s: %v", tag, err)
+		}
+	}
+	for _, tag := range outbounds_strings {
+		outbounds.Add(tag)
+	}
+
+	log.Printf("SSH Service starting 0.0.0.0:%s\n", sshutil.SSHPORT)
```

Comment:
The `0.0.0.0` is not very useful in the message.

```suggestion
	log.Printf("SSH service starting on port %s\n", sshutil.SSHPORT)

```